ENIGMAsystem/SHELL/Graphics_Systems/General/GSbackground.cpp

```diff
@@ -183,45 +192,48 @@ void draw_background_general(int back, gs_scalar left, gs_scalar top, gs_scalar
 
 void draw_background_tiled(int back, gs_scalar x, gs_scalar y, int color, gs_scalar alpha)
 {
-    alpha=CLAMP_ALPHAF(alpha);
-    get_background(bck2d,back);
-    x = (x<0?0:bck2d->width)-fmod(x,bck2d->width);
-    y = (y<0?0:bck2d->height)-fmod(y,bck2d->height);
+  alpha=CLAMP_ALPHAF(alpha);
+  const Background& bck2d = backgrounds.get(back); 
+  const TexRect& tr = bck2d.textureBounds;
+  
+  x = (x<0?0:bck2d.width)-fmod(x,bck2d.width);
+  y = (y<0?0:bck2d.height)-fmod(y,bck2d.height);
 
-  const gs_scalar tbx = bck2d->texturex, tby = bck2d->texturey,
-                  tbw = bck2d->texturew, tbh = bck2d->textureh;
+  const gs_scalar tbx = tr.x, tby = tr.y,
+                  tbw = tr.w, tbh = tr.h;
 
-  const int hortil = int (ceil((gs_scalar)room_width/((gs_scalar)bck2d->width))) + 1,
-            vertil = int (ceil((gs_scalar)room_height/((gs_scalar)bck2d->height))) + 1;
+  const int hortil = int (ceil((gs_scalar)room_width/((gs_scalar)bck2d.width))) + 1,
+            vertil = int (ceil((gs_scalar)room_height/((gs_scalar)bck2d.height))) + 1;
 
-  gs_scalar xvert1 = -x, xvert2 = xvert1 + bck2d->width, yvert1, yvert2;
+  gs_scalar xvert1 = -x, xvert2 = xvert1 + bck2d.width, yvert1, yvert2;
   for (int i=0; i<hortil; ++i)
   {
-    yvert1 = -y; yvert2 = yvert1 + bck2d->height;
+    yvert1 = -y; yvert2 = yvert1 + bck2d.height;
     for (int c=0; c<vertil; ++c)
     {
-      draw_primitive_begin_texture(pr_trianglestrip, bck2d->texture);
+      draw_primitive_begin_texture(pr_trianglestrip, bck2d.textureID);
       draw_vertex_texture_color(xvert1,yvert1,tbx,tby,color,alpha);
       draw_vertex_texture_color(xvert2,yvert1,tbx+tbw,tby,color,alpha);
       draw_vertex_texture_color(xvert1,yvert2,tbx,tby+tbh,color,alpha);
       draw_vertex_texture_color(xvert2,yvert2,tbx+tbw,tby+tbh,color,alpha);
       draw_primitive_end();
       yvert1 = yvert2;
-      yvert2 += bck2d->height;
+      yvert2 += bck2d.height;
     }
     xvert1 = xvert2;
-    xvert2 += bck2d->width;
+    xvert2 += bck2d.width;
   }
 }
 
 void draw_background_tiled_ext(int back, gs_scalar x, gs_scalar y, gs_scalar xscale, gs_scalar yscale, int color, gs_scalar alpha, bool htiled, bool vtiled)
 {
   alpha=CLAMP_ALPHAF(alpha);
-	get_background(bck2d,back);
+	const Background& bck2d = backgrounds.get(back); 
+  const TexRect& tr = bck2d.textureBounds;
```

Comment:
Indentation inconsistencies all over the place. Two spaces ftw.

Reply:
I only did a find and replace. indent was messed before I got here

Reply:
The single line you deleted was consistent with itself though, not really an excuse for being inconsistent with all new lines you've added.